### PR TITLE
fix(form/zod-resolver): fix error message priority

### PR DESCRIPTION
**Pull request by GuiLec**
This PR is to solve the following issue: https://github.com/mantinedev/mantine/issues/5411

### CODE REVIEW

File: packages/@mantine/form/src/resolvers/zod-resolver/zod-resolver.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export function zodResolver<T extends Record<string, any>>(schema: ZodSchema<T>)
 
     const results: Record<string, any> = {};
 
-    (parsed as ZodParseError).error.errors.forEach((error) => {
+    (parsed as ZodParseError).error.errors.toReversed().forEach((error) => {
       results[error.path.join('.')] = error.message;
```
Comment (GuiLec) [inline packages/@mantine/form/src/resolvers/zod-resolver/zod-resolver.ts:34]: If a field has multiple errors, the first ones are overwritten by the last one. Reversing errors order makes sure that the first error message is returned in the result

**Comment by rtivital** [comment]
`zodResolver` is `@mantine/form` package was deprecated in the previous minor version, if you need to fix this issue, send a PR to https://github.com/mantinedev/mantine-form-zod-resolver

**Comment by rtivital** [comment]
I also think that it is opinionated and should be configurable with resolver option.

**Comment by GuiLec** [comment]
> `zodResolver` is `@mantine/form` package was deprecated in the previous minor version, if you need to fix this issue, send a PR to https://github.com/mantinedev/mantine-form-zod-resolver

@rtivital done here: https://github.com/mantinedev/mantine-form-zod-resolver/pull/1

**Pull request closed without merging**